### PR TITLE
[omnilink] Omnilink Binding initial contribution

### PR DESCRIPTION
**Pull request by byrdman7735**

### CODE REVIEW

File: shippable.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+language: java
```
Comment (martinvw) [inline shippable.yml]: I don't think we want this merged, or you have to do sales pitch

File: addons/pom.xml [hunk 2]
```diff
@@ -19,7 +19,7 @@
     <module>binding</module>
     <module>io</module>
     <module>ui</module>
-    <module>voice</module>
+<!--    <module>voice</module> -->
```
Comment (martinvw) [inline addons/pom.xml]: Please revert this

File: addons/binding/pom.xml [hunk 3]
```diff
@@ -74,6 +74,7 @@
     <module>org.openhab.binding.yamahareceiver</module>
     <module>org.openhab.binding.zoneminder</module>
     <module>org.openhab.binding.zway</module>
+    <module>org.openhab.binding.omnilink</module>
```
Comment (martinvw) [inline addons/binding/pom.xml]: Add in the alphabetical right place.

**Comment by martinvw** [comment]
@byrdman7735 thanks your contribution!

Some really early remarks:

* It seems you are referring to 2.1.0-SNAPSHOT from your pom but we recently updated to 2.2.0-SNAPSHOT. 
* Please follow the PR template and add some information where this PR is about.
* Please add a readme as described in guidelines / archetype
* Remove the `.DS_Store` you added, see: https://github.com/openhab/openhab2-addons/pull/2592/files
* The copyright headers seem to be outdated, can you fix all of them by running `mvn license:format` 
* Please add docblocks to all classes including their author and purpose.
* Please squash your commits already as far as possible (maybe back to the merge commit)

Please also check:

http://docs.openhab.org/developers/development/bindings.html
http://docs.openhab.org/developers/development/guidelines.html

It would be great if you could squash your commits and change commit message to include a `Signed-off-by` (to certify that you wrote it or otherwise have the right to pass it on as an open-source), see http://docs.openhab.org/developers/contributing/contributing.html#sign-your-work

Although in this case it's most likely troublesome because you merged in the master, in this project we use rebasing and squashing to create a sequential history. If you are not able to do it yourselves then we will help you after everything is reviewed and completed 😄 

For squashing / rebasing see:
http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

**Comment by martinvw** [comment]
@byrdman7735 could you give us a status update?

**Comment by byrdman7735** [comment]
Sorry please have someone else take it over.

Rob Byrd, MCSE MCSA
Byrd Computer Services LLC
585-210-2556
www.Byrdcs.com

**Comment by martinvw** [comment]
@kaikreuzer there is not yet a proper sign off what can @byrdman7735 do such that someone can take over his work without any legal risks?

**Comment by byrdman7735** [comment]
I did not add or change any of the code so there is none of my work.

Rob Byrd, MCSE MCSA
Byrd Computer Services LLC
585-210-2556
www.Byrdcs.com

**Comment by byrdman7735** [comment]
I don't know I just clicked the download button.

**Comment by martinvw** [comment]
Indeed and this PR is originating from the repo of @QuailAutomation which even contains very recent commits in the openhab2-addons fork by indeed @craigham

So it would be great if you could come out of the shadows and inform us about the status of this PR 😃

**Comment by craigham** [comment]
@martinvw   I have recently been trying to get my master caught up with the openhab master and been having some issues getting it to build. 
 
I would add if there is anyone that would be willing to do the work which would be required to get this binding so it can be added to the main master I would very much appreciate the help.  Time constraints make it difficult for me to get enough time to address everything needed for acceptance.

That being said, I will continue plugging away.

**Comment by craigham** [comment]
@martinvw I have been doing some work on this and have a better idea now of what is going on.  

Last year myself along with Digital Dan and boc-tothefuture started working on this binding.  Unsolicited by myself, byrdman7735 created this PR request.  Unfortunately, he requested a merge from an incorrect branch on my fork.  For some background, I had originally done some work on this binding, but then when Dan started helping out, we created another branch because we largely re-did my previous work to better match best practices which Dan taught us.  That branch is called: 'digitaldan-omnilink-binding-thing-types'.  understandable how byrdman mistakenly chose the incorrect branch, the naming wasn't obvious.  I was away from my house which had the omnilink for about 4 months, and I somehow missed that this PR request was created and in turn you guys started spending resources on it.  

I really would like to get this binding into the openhab master, it seems fairly complete and numerous people have been using it.

I will look at your comments and make the adjustments you suggested.  However, to move this forward how would you suggest we proceed?  Should we close this PR, and instead create one based on the correct branch over on my forked repository?

**Comment by craigham** [comment]
@martinvw I have another question.  I now have my master caught up to the openhab master.  However, I am unsure what I should be doing with the Dev branch we have been using for the omnilink binding.  It has around 300 commits I believe.  I was reading about rebasing it to the current master, but will that cause issues because others have been collaborating off this branch of mine?  Thanks.

**Comment by martinvw** [comment]
Sorry, I kind of missed your question. 🤔

Rebasing while others are also contributing via your branch is indeed not optimal. But maybe it is good to take this pain now and use that moment to squash the commits into a single one and rebase it now. Other contributors from your branch can then either rebase on your branch or just throw away their copy of your branch and check out a fresh copy. 

If it’s not clear just let me know I promise to respond quicker this time 😉

**Comment by martinvw** [comment]
and yes feel free to open a new one and refer to it from here, if needed I can close this one.

**Comment by martinvw** [comment]
@craigham what is the status of this PR, did you succeed in moving around the branches?

Is it ready to be reviewed? It currently does not seem to build on our servers I don't know whether you were aware of that?

**Comment by craigham** [comment]
@martinvw  I was able to move the code to a more recent version of upstream openhab.

I would say this PR should be killed.  

As for getting this code merged in I'm not sure how to proceed.  There was contributions by a few people, without really keeping track, so things like specifying author, and class documentation will be tough.

**Comment by martinvw** [comment]
> As for getting this code merged in I'm not sure how to proceed. There was contributions by a few people, without really keeping track, so things like specifying author, and class documentation will be tough.

I assume git blame would identify who wrote what or where files not correctly moved around? Class documentation should indeed be there but most of the time that is not the biggest problem.

> I would say this PR should be killed.

So you mean you want to create a new PR from another branch? Feel free to do anytime 👍

**Comment by digitaldan** [comment]
Hi @craigham ,  just noticed this as I ripped out the existing DSC system in our new house this week and replaced it with a Omnilink system, and had just started going back through the code.  Did not realize, or totally forgot there was a PR open for our binding.

I think the majority of the code was contributed by only a few of us early on, would it be enough to add you , @boc-tothefuture  and myself and any others i am not remembering  to most of the files as the initial contributors ?   I’m not sure what has been added since i was last working on this, but at first glance it does not seem that much different.   

I’m now committed to this binding in a major way now :-) so I am happy to help get it submitted.

**Comment by craigham** [comment]
Hi Dan,

That is great news for the sake of the binding that you have installed Omnilink again, and good to have you back.

As for authorship that sounds perfectly fine to me.  I really don't care about recognition/etc, just would like a binding that works, and has some people that can keep it updated.

Also, I have done a fair bit of work on 3 bindings, but have never gone through the process to get them merged into master.  If you could help with that, that would be awesome.  I imagine the hardest part will be documentation requirements, but if you and hopefully @boc-tothefuture  could chip in, the task doesn't sound so daunting.

Because the branch got so old, I had a hard time rebasing it to recent code, so instead I basically copied what we had come up with for binding code, and applied it to a more recent master.  That new branch is omnilink-binding, not the old digital dan one.

Do you think you could take a look and suggest what would be good steps to get this finally merged in?

cheers,
Craig

**Comment by boc-tothefuture** [comment]
I would think that at a minimum we need to document the binding and convert to the new build system?  @digitaldan Have you converted any of your previous bindings to the new build system?

**Comment by digitaldan** [comment]
I have not used the new build system yet, but @wborn just converted once of my bindings (logitech harmony), so i take a look at that.  I can also help with the documentation and other submission issues.

**Comment by craigham** [comment]
@boc-tothefuture Do you have a link handy which describes the new build system?  I didn't know about it.

thanks

**Comment by J-N-K** [comment]
Have a look here: https://github.com/openhab/openhab2-addons/issues/5005

**Comment by digitaldan** [comment]
@craigham @boc-tothefuture Couple things, first who is going to tackle the build changes?  I don't mind doing it, i have a few bindings i need to move over as well, but i don't want to step on anyones toes if someone else has already started

Going through the code i see a couple of areas that need some minor cleanup, so i'll start submitting PR's later today or tomorrow.  

last is that i think we should close this PR, work a bit on the binding and open a new one as talked about above.  Sound ok?

**Comment by boc-tothefuture** [comment]
I am good with closing the PR and coming backwhen we have something.  If you want to tackle the build changes, that would be great.  Right now I am working on making the unit support more generic and supporting on_for_seconds, on_for_minutes, on_for_hours.  We have had multiple requests in the forums to support different unit types.

**Comment by craigham** [comment]
I agree we should close this PR.

I am jumping on a plane in 3 hours for a 15 day vacation, so I won't be able to contribute until I get back.  I should be able to merge PRs while away though.

**Comment by digitaldan** [comment]
what branch/repo are we working off of?

**Comment by craigham** [comment]
I think work should now be on the omnilink-binding branch.  Awhile ago I caught it up to the openhab master, and then reapplied our code to it.

**Comment by digitaldan** [comment]
Sounds good, I will close this now.  Have fun on your trip!

**Pull request closed without merging**